src/Core/src/Handlers/Image/ImageHandler.iOS.cs

```diff
@@ -38,11 +35,16 @@ public static void MapBackground(IImageHandler handler, IImage image)
 		public static void MapSource(IImageHandler handler, IImage image) =>
 			MapSourceAsync(handler, image).FireAndForget(handler);
 
-		public static Task MapSourceAsync(IImageHandler handler, IImage image) =>
-			handler.SourceLoader.UpdateImageSourceAsync();
+		public static async Task MapSourceAsync(IImageHandler handler, IImage image) =>
+			await handler.SourceLoader.UpdateImageSourceAsync();
+		
+		void IImageSourcePartSetter.SetImageSource(UIImage? source)
+		{
+			PlatformView.Image = source;
 
-		void IImageSourcePartSetter.SetImageSource(UIImage? obj) =>
-			PlatformView.Image = obj;
+			if (VirtualView.Source is IStreamImageSource)
+				PlatformView.InvalidateMeasure(VirtualView);
```

Comment:
This is the main change.

Reply:
Did we do this before? Why is this suddenly needed now?

Reply:
Can we add a test for this to make sure this doesn't happen again?

Reply:
No, we didn't use it before. I have added a device test.